### PR TITLE
feat: allow admins to edit all resources 📝

### LINKED ISSUE #267: Allow admins to edit all resources ✍️
### Description

In order to ensure the highest quality around our resources, we will allow admins to edit resources (ie: title, description, tags). We also have a ColorStack ambassador who will be moderating/updating content as well.

### Acceptance Criteria

We need to introduce a notion of linking admins and members (b/c some admins have member accounts). We'll need to add a field such as `member_id` on the `admins` table, which references the `students` table.

Then, when on the `/resources` page, we should make the [`editable`](https://github.com/colorstackorg/oyster/blob/main/apps/member-profile/app/routes/_profile.resources.tsx#L131) true if the resource was either posted by the logged-in user or if the logged-in user is an admin (will need a query for this).

### PR DESCRIPTION
**Pull request by nicholasg2001**
### Migrations:
- `20240614123146_add_member_id_admin.ts`: Inserts member_id column in admins table and makes it a foreign key on students table id field.
- `20240614185515_populate_member_id_in_admins_table.ts`: Updates existing admins new member_id column with their corresponding students.id record. I translated the kysely code in this file into raw sql and ran it in the Postgres CLI and it worked fine. I am super stumped as to why the migration file is not functioning properly at the moment.

image

This is a test resource with the editable icon showing while **logged in as an admin.**

Closes #267 

## Type of Change 🐞

- [x] Feature - A non-breaking change which adds functionality.
- [ ] Fix - A non-breaking change which fixes an issue.
- [ ] Refactor - A change that neither fixes a bug nor adds a feature.
- [ ] Documentation - A change only to in-code or markdown documentation.
- [ ] Tests - A change that adds missing unit/integration tests.
- [ ] Chore - A change that is likely none of the above.

## Checklist ✅

- [x] I have done a self-review of my code.
- [x] I have manually tested my code (if applicable).
- [ ] I have added/updated any relevant documentation (if applicable).

### CODE REVIEW

File: packages/db/src/migrations/20240614185515_populate_member_id_in_admins_table.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  await db
+    .updateTable('admins')
+    .set({
+      member_id: db
+        .selectFrom('students')
+        .select(['id'])
+        .where('students.email', '=', 'admins.email'),
+    })
+    .execute();
+}
```
Comment (tomas-salgado) [inline packages/db/src/migrations/20240614185515_populate_member_id_in_admins_table.ts:14]: We do want to try to have just one migration per PR, so it's fine to combine this with the `up` function in the other migration (and similarly with the `down` function)

File: packages/db/src/migrations/20240614123146_add_member_id_admin.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    })
+    .execute();
+
+  await db
+    .updateTable('admins')
+    .set({
+      member_id: db
+        .selectFrom('students')
+        .select(['id'])
+        .where('students.email', '=', 'admins.email'),
```
Comment (tomas-salgado) [inline packages/db/src/migrations/20240614123146_add_member_id_admin.ts:17]: Here we want to use `.whereRef` rather than `.where`

**Review by tomas-salgado — CHANGES_REQUESTED** [review]
Great work @nicholasg2001! Just one minor change, but everything else looks good!

File: packages/db/src/migrations/20240614123146_add_member_id_admin.ts (hunk 2 above)
Comment (nicholasg2001) [reply packages/db/src/migrations/20240614123146_add_member_id_admin.ts:17]: fixed!

**Review by tomas-salgado — APPROVED** [review]
Great stuff! @ramiAbdou will take a final look before we merge but LGTM! 🥳

**Review by ramiAbdou — APPROVED** [review]

**Pull request merged**